File: Contests/Codechef/MAY20A/CHANDF.cpp
#include <bits/stdc++.h>
using namespace std;

#define ll long long
#define pb push_back
#define mp make_pair
#define ar array

int main() {
	ios_base::sync_with_stdio(0); cin.tie(0);

#ifndef ONLINE_JUDGE
    // for getting input from input.txt
    freopen("input.txt", "r+", stdin);
    // for writing output to output.txt
    freopen("output.txt", "w+", stdout);
#endif

    int t = 1;
    while(t--) {
    	ll x,y,l,r;
    	x = 400;
    	y = 963;
    	l = 0;
    	r = (x|y);
    	for(int i=r;i>=0;i--) {
    		ll val = -1;
    		ll ans = 0;
    		for(int j=l;j<=i;j++) {
    			ll tmp = (x & j) * (y & j);
    			if(tmp > val) {
    				val = tmp;
    				ans = j;
    			}
    		}
    		ll ans2 = (x|y) & i;
    		if(ans2 != ans) {
    			cout << i << endl;
    		}
    	}
    }
}

File: Contests/Codechef/MAY20A/CHANDF.py
import sys 
# For getting input from input.txt file 
sys.stdin = open('input.txt', 'r')  
  
# Printing the Output to output.txt file 
sys.stdout = open('output.txt', 'w') 

def binary(n):
	binn = bin(n)[2:]
	while(len(binn) < 25):
		binn = '0' + binn
	return binn

t = int(input())
for _ in range(t):
	x,y,l,r = map(int,input().split())
	val = -1
	ans = 0
	for i in range(l,r+1):
		tmp = (x & i) * (y & i)
		if tmp > val:
			ans = i
			val = tmp
	print(ans)
	binx = binary(x)
	biny = binary(y)
	bina = binary(ans)
	print("x",binx,x)
	print("y",biny,y)
	print("o",binary(x | y),(x|y))
	print("r",binary(r),r)
	print("a",bina,ans)
	print("e",binary((x|y)&r),((x|y)&r))
	print((x|y) - ans)
	print((x|y) - r)